xonsh/main.py

```diff
@@ -236,6 +236,9 @@ def start_services(shell_kwargs):
     env = builtins.__xonsh_env__
     rc = shell_kwargs.get('rc', None)
     rc = env.get('XONSHRC') if rc is None else rc
+    if shell_kwargs['shell_type'] != 'none':
```

Comment:
We set `XONSH_INTERACTIVE` in `premain` after we've returned the `env` from this function, so I just went with this -- can also rework `premain` a bit